keyboard_handler/include/keyboard_handler/keyboard_handler_unix_impl.hpp

```diff
@@ -38,11 +39,21 @@ class KeyboardHandlerUnixImpl : public KeyboardHandlerBase
   using tcgetattrFunction = std::function<int (int, struct termios *)>;
   using tcsetattrFunction = std::function<int (int, int, const struct termios *)>;
   using readFunction = std::function<ssize_t(int, void *, size_t)>;
+  using signal_handler_type = void (*)(int);
 
   /// \brief Default constructor
   KEYBOARD_HANDLER_PUBLIC
   KeyboardHandlerUnixImpl();
 
+  /// \brief Constructor with option to not install signal handler for SIGINT
+  /// \param install_signal_handler if true signal handler for SIGINT will be installed,
+  /// otherwise not.
+  /// \note In case if install_signal_handler is false caller code should call static
+  /// KeyboardHandlerUnixImpl::restore_buffer_mode_for_stdin() in case of process termination
+  /// caused by signal arrival.
+  KEYBOARD_HANDLER_PUBLIC
+  explicit KeyboardHandlerUnixImpl(bool install_signal_handler);
```

Comment:
should a similar constructor be added to the windows implementation just for compatibility?
Even if it never installs a signal handler, having the same interface in both will be helpful for users.

Reply:
I've considered an option to add it to the  Windows version but decided to not to do this.
And here is why:
The rational for having it in both versions is to avoid construction like this:
```
KeyboardHandler keyboard_handler;
#ifndef _WIN32
  keyboard_handler = KeyboardHandler(false);
#else
  keyboard_handler = KeyboardHandler();
#endif
```
but it's actually not enough if I would add this constructor to the Windows version will need to workaround with `ifndef _WIN32` following construct
```
#ifndef _WIN32
context->add_on_shutdown_callback(KeyboardHandler::restore_buffer_mode_for_stdin());
#endif
```
or add this "dummy" `static bool restore_buffer_mode_for_stdin();` method to the Windows version as well.
Which is too much and will look very odd from API perspective and design.

I think it will be Ok to have this extra constructor with parameter only in Unix/POSIX version since:
1. It's relevant only to the Unix/POSIX version
2. Default constructor will properly handle situation in case of presence native signal handler from rclcpp context via signal chaining feature.
3. If for some reasons someone will need to use KeyboardHandler without it's internal signal handler there are alway available option to call it explicitly with `#ifndef _WIN32` guard.

Reply:
> or add this "dummy" static bool restore_buffer_mode_for_stdin(); method to the Windows version as well.

Yeah, that was the alternative I was thinking.
The current approach looks fine though.